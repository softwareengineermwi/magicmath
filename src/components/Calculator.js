import React, { useState } from 'react';
import Input from './Input';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setState({ ...state, ...calculate(state, e.target.innerText) });
  };

  const { next, total } = state;

  return (
    <div className="grid container" data-testid="mycalc">
      <div className="col-6">
        <h3>Let`s do some math</h3>
      </div>

      <div className="col-6 grid container max-width-sm">
        <div className="col-12 bg-contrast-low text-right padding-md">
          <input className="resultscreen" value={next || total || screen || 0} />
        </div>

        <Input handler={handleClick} />
      </div>
    </div>
  );
};

export default Calculator;
