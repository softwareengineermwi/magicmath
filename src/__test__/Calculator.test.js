import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

describe('Test Calculator Components', () => {
  it('Should render without error', () => {
    const divElement = document.createElement('div');
    ReactDOM.render(<Calculator />, divElement);
  });

  it('Should render correctly', () => {
    const { getByTestId } = render(<Calculator />);
    expect(getByTestId('mycalc')).toHaveTextContent('Let`s do some math');
  });

  it('Should match snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});