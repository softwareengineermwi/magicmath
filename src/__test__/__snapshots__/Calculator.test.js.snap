// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator Components Should match snapshot 1`] = `
<div
  className="grid container"
  data-testid="mycalc"
>
  <div
    className="col-6"
  >
    <h3>
      Let\`s do some math
    </h3>
  </div>
  <div
    className="col-6 grid container max-width-sm"
  >
    <div
      className="col-12 bg-contrast-low text-right padding-md"
    >
      <h4>
        0
      </h4>
    </div>
    <div
      className="grid col-12"
    >
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md foo"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md foo"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md foo"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md foo"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="col-6 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="col-3 btn btn--subtle padding-y-md foo"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
